### ISSUE TITLE
Properties named 'value'

### ISSUE THREAD

**Issue by rhythnic** [erikras/redux-form#477]
I have some nested objects in a deep from that have the property 'value' in the object.  For example:
```
{
  details: [
    {
      name: 'detail name',
      value: 'detail value'
    }
  ]
}
```

It seems to be causing some unexpected behavior, so far with resetForm and getValues.

When using resetForm, all the properties inside of the objects with a 'value' key are set to undefined, despite having been initialized successfully with a value.

When using getValues, the 'details' array, which should look like it does above, looks like:
```
{
  details: [
    {
      initial: 'detail value',
      value: 'detail value'
    }
  ]
}
```

**Comment by erikras** [comment]
Good catch. Will investigate. :mag:

**Comment by erikras** [comment]
Fix published in [`v4.0.7`](https://github.com/erikras/redux-form/releases/tag/v4.0.7).
